src/helm/benchmark/run_specs/medhelm/benchmark_config.py

```diff
@@ -0,0 +1,201 @@
+import cattrs
+import yaml
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+from abc import ABC
+
+from helm.benchmark.annotation.annotator import AnnotatorSpec
+from helm.benchmark.annotation.model_as_judge import AnnotatorModelInfo
+from helm.benchmark.metrics.metric import MetricSpec
+from helm.benchmark.metrics.common_metric_specs import (
+    get_basic_metric_specs,
+    get_exact_match_metric_specs,
+    get_f1_metric_specs,
+    get_language_modeling_metric_specs,
+    get_classification_metric_specs,
+    get_summarization_metric_specs,
+)
+from helm.common.gpu_utils import get_torch_device_name
+
+
+@dataclass(frozen=True)
+class MetricConfig(ABC):
+    """Base class for all metric configurations"""
+
+    pass
+
+
+@dataclass(frozen=True)
+class SimpleMetricConfig(MetricConfig):
+    """Configuration for simple string-based metrics like 'exact_match'"""
+
+    name: str
+
+
+@dataclass(frozen=True)
+class JuryMetricConfig(MetricConfig):
+    """Configuration for jury-based metrics with multiple judges"""
+
+    prompt_file: str
+    judges: List[AnnotatorModelInfo]
+
+
+@dataclass(frozen=True)
+class BenchmarkConfig:
+    """
+    A benchmark configuration is an immutable data structure that holds
+    the configuration for a specific benchmark, including prompt, dataset and metric
+    """
+
+    name: str
+    """Name of the benchmark"""
+
+    description: str
+    """Description of the benchmark"""
+
+    prompt_file: str
+    """Path to the prompt file. This prompt will be used for all instances of the benchmark."""
+
+    dataset_file: str
+    """Path to the dataset file. This dataset will be used to populate the context in the prompt."""
+
+    metrics: List[MetricConfig]
```

Comment:
Note that doing this means that you can't roundtrip to YAML. If you need this, there there are a couple ways to get this to work

Switch to putting everything in the same class:
```python
@dataclass(frozen=True)
class MetricConfig:
    name: str
    prompt_file: str
    judges: List[AnnotatorModelInfo]
```

Use optionals:
```python
@dataclass(frozen=True)
class MetricConfig:
    simple: Optional[SimpleMetricConfig] = None
    judge: Optional[JudgeMetric] = None
```

Use a Union - this _only_ works if your dataclasses have different fields, so that the serializer can figure out which class to use based on the fields:
```python
    metrics: List[Union[SimpleMetricConfig, JudgeMetricConfig]]
```